### PR TITLE
DS-4: Component library

### LINKED ISSUE #4: DS - Project
- Create project
- Add links to docs

### PR DESCRIPTION
**Pull request by Zimtir**
- Closes #4

### CODE REVIEW

File: development/components/.gitignore [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# misc
+.DS_Store
+.env.local
+.env.development.local
+.env.test.local
+.env.production.local
+
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
```
Comment (Proza1k) [inline development/components/.gitignore:23]: add empty line

File: development/components/README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
+This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
```
Comment (Proza1k) [inline development/components/README.md:1]: update file description

File: development/components/.storybook/main.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+const path = require('path');
```
Comment (Proza1k) [inline development/components/.storybook/main.js:1]: is there a setting for typescript?

File: development/components/src/components/Button.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import React, { FC } from 'react';
+
+interface ButtonProps {
+  /**
+   * Simple click handler
```
Comment (Proza1k) [inline development/components/src/components/Button.tsx:5]: remove please comments

File: development/components/src/components/Button.tsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+interface ButtonProps {
+  /**
+   * Simple click handler
+   */
+  onClick?: () => void;
+}
+
+/**
+ * The world's most _basic_ button
```
Comment (Proza1k) [inline development/components/src/components/Button.tsx:11]: same

File: development/components/src/index.tsx [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import './index.css';
+import App from './App';
+import * as serviceWorker from './serviceWorker';
+
+ReactDOM.render(<App />, document.getElementById('root'));
+
+// If you want your app to work offline and load faster, you can change
```
Comment (Proza1k) [inline development/components/src/index.tsx:9]: remove please comments

File: development/components/src/react-app-env.d.ts [hunk 7]
```diff
@@ -0,0 +1 @@
+/// <reference types="react-scripts" />
```
Comment (Proza1k) [inline development/components/src/react-app-env.d.ts:1]: Is this file being used somewhere? If not, then delete

File: development/components/package.json [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2",
+    "react-scripts": "4.0.3",
+    "typescript": "~3.7.2"
+  },
+  "scripts": {
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "test": "react-scripts test",
+    "eject": "react-scripts eject",
```
Comment (Proza1k) [inline development/components/package.json:28]: Может стоит добавить eslint сюда?

File: development/components/src/serviceWorker.ts [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+// register() is not called by default.
+
+// This lets the app load faster on subsequent visits in production, and gives
+// it offline capabilities. However, it also means that developers (and users)
+// will only see deployed updates on subsequent visits to a page, after all the
+// existing tabs open on the page have been closed, since previously cached
+// resources are updated in the background.
+
+// To learn more about the benefits of this model and instructions on how to
+// opt-in, read https://bit.ly/CRA-PWA
```
Comment (Proza1k) [inline development/components/src/serviceWorker.ts:11]: remove comments please

File: development/components/src/setupTests.ts [hunk 10]
```diff
@@ -0,0 +1,5 @@
+// jest-dom adds custom jest matchers for asserting on DOM nodes.
+// allows you to do things like:
+// expect(element).toHaveTextContent(/react/i)
+// learn more: https://github.com/testing-library/jest-dom
```
Comment (Proza1k) [inline development/components/src/setupTests.ts:4]: remove please comments

**Review by Proza1k — CHANGES_REQUESTED** [review]
также думаю стоит добавить .prettier конфиг для формата документов

File: development/components/package.json (hunk 8 above)
Comment (Zimtir) [reply development/components/package.json:28]: Add an issue into list of backlog

File: development/components/.editorconfig [hunk 11]
```diff
@@ -0,0 +1,17 @@
+# http://EditorConfig.org
```
Comment (Proza1k) [inline development/components/.editorconfig:1]: remove comment

File: development/components/.eslintrc.json [hunk 12]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (Proza1k) [inline development/components/.eslintrc.json:1]: format code

File: development/components/.gitignore (hunk 1 above)
Comment (Proza1k) [reply development/components/.gitignore:23]: add empty line

File: development/components/README.md [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
```
Comment (Proza1k) [inline development/components/README.md:1]: please update information readme

File: development/components/public/index.html [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
+    <meta name="description" content="Web site created using create-react-app" />
+    <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
+    <!--
+      manifest.json provides metadata used when your web app is installed on a
+      user's mobile device or desktop. See https://developers.google.com/web/fundamentals/web-app-manifest/
+    -->
```
Comment (Proza1k) [inline development/components/public/index.html:13]: remove comments

File: development/components/public/index.html [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+    <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
+    <!--
+      Notice the use of %PUBLIC_URL% in the tags above.
+      It will be replaced with the URL of the `public` folder during the build.
+      Only files inside the `public` folder can be referenced from the HTML.
+
+      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
```
Comment (Proza1k) [inline development/components/public/index.html:23]:
```suggestion
```

File: development/components/public/index.html [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+    <!--
+      This HTML file is a template.
+      If you open it directly in the browser, you will see an empty page.
+
+      You can add webfonts, meta tags, or analytics to this file.
+      The build step will place the bundled scripts into the <body> tag.
+
+      To begin the development, run `npm start` or `yarn start`.
+      To create a production bundle, use `npm run build` or `yarn build`.
+    -->
```
Comment (Proza1k) [inline development/components/public/index.html:38]:
```suggestion
```

**Review by Proza1k — CHANGES_REQUESTED** [review]

File: development/components/.gitignore (hunk 1 above)
Comment (Proza1k) [reply development/components/.gitignore:23]:
```suggestion
yarn-error.log*

package-lock.json
yarn.lock
```

**Review by Proza1k — APPROVED** [review]

**Pull request merged**